ci/run_envoy_docker.sh

```diff
@@ -18,7 +18,8 @@ USER_GROUP=root
 
 mkdir -p "${ENVOY_DOCKER_BUILD_DIR}"
 # Since we specify an explicit hash, docker-run will pull from the remote repo if missing.
-docker run --rm -t -i -u "${USER}":"${USER_GROUP}" -v "${ENVOY_DOCKER_BUILD_DIR}":/build \
+docker run --rm -t -i -e http_proxy=${http_proxy} -e https_proxy=${https_proxy} -e HTTP_PROXY=${http_proxy} -e HTTPS_PROXY=${https_proxy} \
```

Comment:
Thanks for this fix. Why both upper and lower case proxy vars?

Reply:
Bazel respects uppercase..

https://docs.bazel.build/versions/master/external.html#using-proxies

Reply:
Why set the lowercase then?

Reply:
git uses the lower case one

https://git-scm.com/docs/git-config#git-config-httpproxy

So one for git and another for bazel http/https files if defined.

Reply:
OK, great. Can you update docs in https://github.com/envoyproxy/envoy/blob/master/ci/README.md#building-and-running-tests-as-a-developer and https://github.com/envoyproxy/envoy/blob/master/bazel/README.md on how to use these vars? Thanks!

Reply:
One other though; can't we do this in Bazel? I.e. always use upper case, and only when we are invoking native `git` do the environment variable remap? This might be more Bazel plumbing than its worth.

Reply:
We need to in inject the environment variable, the question is which one.. 
I tested once with lower case and once with upper case, both works.. we can get rid of one of them.. lower case is more famous so we can just keep that.. what do u suggest?